packages/plugin-dom-renderer/DomRenderer.ts

```diff
@@ -25,43 +27,77 @@ export class DomRenderer extends Renderer<DocumentFragment> {
+     *
+     * @param context
+     */
+    renderNode(context: DomRenderingContext): DomRenderingContext {
+        const node = context.currentVNode;
+
+        for (const renderingPredicate of this.renderingPredicates) {
+            const renderingFunction = renderingPredicate(node);
+            if (renderingFunction) {
+                return renderingFunction({ ...context }) as DomRenderingContext;
+            }
+        }
+
+        const tagName = node.is(VElement) ? node.htmlTag : 'UNKNOWN-NODE';
+        const fragment = document.createDocumentFragment();
+        const renderedDomNodes: Node[] = [];
+        const renderedElement = document.createElement(tagName);
+        renderedDomNodes.push(renderedElement);
+        fragment.appendChild(renderedElement);
+
+        // If a node is empty but could accomodate children,
+        // fill it to make it visible.
+        if (!node.hasChildren() && !node.atomic) {
+            const placeholderBr = document.createElement('BR');
+            renderedElement.appendChild(placeholderBr);
+            renderedDomNodes.push(placeholderBr);
+        }
+
+        context.parentNode.appendChild(fragment);
+
+        context.renderingMap = createMapFromDom(
+            renderedDomNodes.map(domNode => {
+                return [domNode, node];
+            }),
+        );
+        return context;
+    }
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
 
-    /**
-     * Return true if `a` has the same format properties as `b`.
-     *
-     * @param a
-     * @param b
-     */
-    _isSameTextNode(a: VNode, b: VNode): boolean {
-        if (isMarker(a) || isMarker(b)) {
-            // A Marker node is always considered to be part of the same text
-            // node as another node in the sense that the text node must not
-            // be broken up just because it contains a marker.
-            return true;
-        } else if (!a.is(CharNode) || !b.is(CharNode)) {
-            // Nodes that are not valid in a text node must end the text node.
-            return false;
-        } else {
-            // Char VNodes are the same text node if they have the same format.
-            const formats = Object.keys({ ...(a as CharNode).format, ...(b as CharNode).format });
-            return formats.every(k => !!(a as CharNode).format[k] === !!(b as CharNode).format[k]);
-        }
+    _renderNode(context: DomRenderingContext): DomRenderingContext {
+        context = this.renderNode({ ...context });
+
+        // Map the parsed nodes to the DOM nodes they represent.
+        Array.from(context.renderingMap.keys()).forEach(domNode => {
+            const renderedVNodes = context.renderingMap.get(domNode);
+            renderedVNodes.forEach((node: VNode, index: number) => {
+                VDocumentMap.set(node, domNode, index);
+            });
+        });
```

Comment:
Should have it's own map. Maybe in another PR?

Reply:
Yes, this is out of scope of this PR.